oak_runtime/src/lib.rs

```diff
@@ -1236,15 +1236,32 @@ impl Runtime {
         // by the current Node, since in general this may be lower than "public untrusted".
         self.validate_can_write_to_label(node_id, label)?;
 
+        let node_type = instance.node_type();
+        let node_privilege = instance.get_privilege();
+
+        // If the new node has any ability to communicate externally, we should make sure that it
+        // has the privilege to downgrade its label to "public untrusted".
+        if instance.external_facing() {
+            let downgraded_label = node_privilege.downgrade_label(label);
+            debug!(
+                "Maximum downgraded label for node {}: {:?}",
+                node_name, &downgraded_label
+            );
+            if !downgraded_label.flows_to(&Label::public_untrusted()) {
```

Comment:
Could this just be `self.validate_can_write_to_label(node_id, &Label::public_untrusted())...`? I think that would read better and it is more intentional as to what's happening?

Reply:
No. The node is not yet registered, so does not yet have a node_id. Using the node_id that is passed in will check whether the creator can write to the label, not the newly created node.